src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -422,30 +425,44 @@ protected final void finalizeGUI(boolean wrapInScrollPane) {
 			openInEditingMode(generatableElement);
 		}
 
-		elementUpdateListener.registerUI(pages.size() > 1 ? parameters : centerComponent);
-		listeningEnabled = true;
-
-		disableUnsupportedFields();
-	}
-
-	private void disableUnsupportedFields() {
 		List<String> exclusions = mcreator.getGeneratorConfiguration()
 				.getUnsupportedDefinitionFields(modElement.getType());
-
 		List<String> inclusions = mcreator.getGeneratorConfiguration()
 				.getSupportedDefinitionFields(modElement.getType());
+		disableUnsupportedFields(this, exclusions, inclusions);
 
+		elementUpdateListener.registerUI(pages.size() > 1 ? parameters : centerComponent);
+		listeningEnabled = true;
+	}
+
+	private static void disableUnsupportedFields(Container source, List<String> exclusions, List<String> inclusions) {
 		if (exclusions != null && inclusions != null) { // can't exclude and include together
 			LOG.warn("Field exclusions and inclusions can not be used at the same time. Skipping them.");
 		} else if ((exclusions != null && !exclusions.isEmpty()) || (inclusions != null && !inclusions.isEmpty())) {
 			Map<Container, List<Component>> includedComponents = new HashMap<>();
+			Map<JEntriesList, Map<Class<?>, Tuple<List<String>, List<String>>>> entryLists = new HashMap<>();
 			for (String entry : Objects.requireNonNullElse(exclusions, inclusions)) {
 				try {
 					Stack<Component> hierarchy = new Stack<>();
-					hierarchy.push(this);
-					for (String next : entry.split("\\.")) {
-						Field field = hierarchy.peek().getClass().getDeclaredField(next);
+					hierarchy.push(source); // add the root component
+					String[] path = entry.split("\\.");
+					for (int i = 0; i < path.length; i++) {
+						Field field = hierarchy.peek().getClass().getDeclaredField(path[i]);
 						if (!Component.class.isAssignableFrom(field.getType())) {
+							// this may target a list of potential child entries of JEntriesList
+							if (hierarchy.peek() instanceof JEntriesList entriesList
```

Comment:
JEntriesList could have a type parameter and define entryList on an abstract class level with a getter. This way the need for use of ParameterizedType/reflection can be avoided.

Reply:
The problem is not the access to `entryList` but the lack of a class argument for superior constructor or at least an example child entry (we can't be sure the list is not empty at that point of time).

Reply:
Also the approach you suggested would bring quite some pain to the item properties PR because a component added in that one uses two different entry lists at the same time (and these are too closely related to be split into separate `JEntriesList`s subclasses), and the way this PR (I mean #3675) changes things there's no [effective] limitation to one `entryList()` per component type, so we can easily access both properties and states lists on the same component.

Reply:
You have introduced another list without any comment on when it is used.

> this contains mapped exclusions/inclusions for each child component type for all entries lists found

This still does not explain 

`Map<JEntriesList, Map<Class<?>, Tuple<List<String>, List<String>>>> entryLists = new HashMap<>();`

without doing a serious deep dive into the code below this line

Reply:
I tried to make comments short and concise, did I fail?

Reply:
You should still cover structure of entryLists in format I suggested above:

> A Map where each JEntriesList key represents one entry list of the container and value is map of classes that represent each ...

You will need to provide continuation there as I could not continue the explanation.

You can also split this into multilines if you think it is better, but I don't think anyone from our dev community can currently quickly understand what is eg. second element in your tuple

Reply:
Would it be more clear the way it is if there was only one list but not tuple of two?

Reply:
Maybe, but hard to say as I did not have time to figure out the current code out yet :D